src/cart/crt_init.c

```diff
@@ -1161,14 +1118,37 @@ crt_na_config_init(bool primary, crt_provider_t provider,
 	struct crt_na_config		*na_cfg;
 	int				rc = 0;
 	int				port = -1;
+	char				*save_ptr = NULL;
+	char				*token = NULL;
+	int				idx = 0;
+	int				count = 0;
 
 	if (provider == CRT_PROV_SM)
 		return 0;
 
 	na_cfg = crt_provider_get_na_config(primary, provider);
-	D_STRNDUP(na_cfg->noc_interface, interface, 64);
-	if (!na_cfg->noc_interface)
-		D_GOTO(out, rc = -DER_NOMEM);
+
+	/* CXI provider requires domain to be set */
+	if (provider == CRT_PROV_OFI_CXI && !domain) {
+		D_ERROR("Domain must be set for CXI provider\n");
+		D_GOTO(out, rc = -DER_INVAL);
+	}
+
+	if (interface) {
+		if (provider == CRT_PROV_OFI_CXI) {
+			D_WARN("Interface '%s' ignored for CXI. Using domain '%s' instead\n",
+			       interface, domain);
```

Comment:
I don't necessarily think we should emit a warning for that.

Reply:
I agree it probably shouldn't be a warning, but I still want it to be mentioned in logs in some way. changed to D_INFO for now.